pkg/cloud/handlers.go

```diff
@@ -60,6 +61,32 @@ func RecordRequestsMiddleware() func(*middleware.Stack) error {
 	}
 }
 
+// LogServerErrorsMiddleware is a middleware that logs server errors received when attempting to contact the AWS API
+// A specialized middleware is used instead of the SDK's built-in retry logging to allow for customizing the verbosity
+// of throttle errors vs server/unknown errors, to prevent flooding the logs with throttle errors
+func LogServerErrorsMiddleware() func(*middleware.Stack) error {
+	return func(stack *middleware.Stack) error {
+		return stack.Finalize.Add(middleware.FinalizeMiddlewareFunc("LogServerErrorsMiddleware", func(ctx context.Context, input middleware.FinalizeInput, next middleware.FinalizeHandler) (output middleware.FinalizeOutput, metadata middleware.Metadata, err error) {
+			output, metadata, err = next.HandleFinalize(ctx, input)
+			if err != nil {
+				var apiErr smithy.APIError
+				if errors.As(err, &apiErr) {
+					if _, isThrottleError := retry.DefaultThrottleErrorCodes[apiErr.ErrorCode()]; isThrottleError {
+						// Only log throttle errors under a high verbosity as we expect to see many of them
```

Comment:
I would rather not change our current customer-facing log level for throttle errors in this PR, if the RecordRequest middleware has already been logging those requests at default verbosity. 

Customer may want to increase rate-limits instead of EBS CSI Driver silently taking longer to manage volumes. 

(On the other hand, customer can already alarm on rate-limit metric)

Reply:
The problem with this is that it will flood the error logs on bursty workloads with entirely benign error messages. Treating a retryable throttling response the same as a "VolumeCapacityExceeded" or "ServerInternal" error greatly reduces the signal to noise ratio of the logs.

Reply:
I agree Connor that the existing logging for throttle errors is too aggressive and support increasing the default log level. 

Drew brings up a good point that we might want to split that out into another PR to make it more explicit. I don't have a strong preference, we can do it here as well by splitting it into a separate commit.

Reply:
> greatly reduces the signal to noise ratio of the logs.

I agree with this.

 My comment was mostly to call out that it's a customer facing change, so we should proceed with some caution and not sneak it in another PR. 

Separate commit is fine and/or a small note in the changelog. Resolving!